src/systems/triggered_publisher/TriggeredPublisher.cc

```diff
@@ -657,20 +780,22 @@ void TriggeredPublisher::DoWork()
       this->newMatchSignal.wait_for(lock, 1s,
         [this]
         {
-          return (this->publishCount > 0) || this->done;
+          return (this->publishCount > 0) || this-> triggerServ || this->done;
         });
 
       if (this->publishCount == 0 || this->done)
+      {
         continue;
+      }
 
-      std::swap(pending, this->publishCount);
-    }
-
-    for (auto &info : this->outputInfo)
-    {
-      for (std::size_t i = 0; i < pending; ++i)
+      if (this->publishCount >0)
       {
-        info.pub.Publish(*info.msgData);
+        std::swap(pending, this->publishCount);
+        PublishMsg(pending);
+      }
+      if (this->triggerServ)
```

Comment:
Same for this. This if-statement can probably go outside the scope for the mutex.